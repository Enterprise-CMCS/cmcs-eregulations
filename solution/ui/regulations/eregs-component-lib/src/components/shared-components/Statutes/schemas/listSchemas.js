const titleSelectorList = {
    ssa: {
        name: "Social Security Act",
        titles: [
            {
                title: "11",
                titleRoman: "XI",
            },
            {
                title: "18",
                titleRoman: "XVIII",
            },
            {
                title: "19",
                titleRoman: "XIX",
            },
            {
                title: "21",
                titleRoman: "XXI",
            },
        ],
    },
};

export { titleSelectorList };
