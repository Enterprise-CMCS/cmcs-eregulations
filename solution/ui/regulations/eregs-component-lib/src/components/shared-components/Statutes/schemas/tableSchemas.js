import {
    houseGovUrl,
    ssaGovUrl,
    statuteCompilationUrl,
    usCodeUrl,
} from "../utils/urlMethods.js";
import { getDateLabel } from "../utils/dateMethods.js";

// placeholder for future schema
const acaSchema = [
    {
        header: {
            title: "Primary Column",
            primary: true,
        },
        body: {
            title: "ACA Title",
            label: "ACA Label",
            name: "ACA Name",
            primary: true,
        },
    },
    {
        header: {
            title: "Secondary Column",
            secondary: true,
            subtitles: ["Subtitle One", "Subtitle Two"],
        },
        body: {
            url: "Link href",
            text: "Link text",
            type: "external", // or "pdf"
            secondary: true,
        },
    },
];

const ssaSchema = [
    {
        header: {
            title: "Statute Citation",
            primary: true,
        },
        body: {
            title: (statute) => `SSA Section ${statute.section}`,
            label: (statute) => `${statute.title} U.S.C. ${statute.usc}`,
            name: (statute) => `${statute.name}`,
            primary: true,
        },
    },
    {
        header: {
            testId: "house-gov-link",
            title: "US Code House.gov",
            secondary: true,
            subtitles: [
                () => "Web Page",
                (columnDates) =>
                    getDateLabel(columnDates?.us_code_house_gov ?? {}),
            ],
        },
        body: {
            url: (statute) => houseGovUrl(statute),
            text: (statute) => `${statute.usc}`,
            type: "external",
            secondary: true,
        },
    },
    {
        header: {
            testId: "statute-compilation-link",
            title: "Statute Compilation",
            secondary: true,
            subtitles: [
                () => "PDF Document",
                (columnDates) =>
                    getDateLabel(columnDates?.statute_compilation ?? {}),
            ],
        },
        body: {
            url: (statute) => statuteCompilationUrl(statute),
            text: (statute) => `Title ${statute.statute_title_roman}`,
            type: "pdf",
            secondary: true,
        },
    },
    {
        header: {
            testId: "us-code-annual-link",
            title: "US Code Annual",
            secondary: true,
            subtitles: [
                () => "PDF Document",
                (columnDates) =>
                    getDateLabel(columnDates?.us_code_annual ?? {}),
            ],
        },
        body: {
            url: (statute) => usCodeUrl(statute),
            text: (statute) => `${statute.usc}`,
            type: "pdf",
            secondary: true,
        },
    },
    {
        header: {
            testId: "ssa-gov-link",
            title: "SSA.gov Compilation",
            secondary: true,
            subtitles: [
                () => "Web Page",
                (columnDates) =>
                    getDateLabel(columnDates?.ssa_gov_compilation ?? {}),
            ],
        },
        body: {
            url: (statute) => ssaGovUrl(statute),
            text: (statute) => `${statute.section}`,
            type: "external",
            secondary: true,
        },
    },
];

export { acaSchema, ssaSchema };
