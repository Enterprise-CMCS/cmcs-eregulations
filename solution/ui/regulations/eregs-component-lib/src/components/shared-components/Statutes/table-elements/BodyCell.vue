<script setup>
defineProps({
    cellData: {
        type: Object,
        required: true,
    },
    statute: {
        type: Object,
        required: true,
    },
});
</script>

<template>
    <td
        class="row__cell row__cell--body"
        :class="{
            'row__cell--primary': cellData.header.primary,
            'row__cell--secondary': cellData.header.secondary,
        }"
    >
        <template v-if="cellData.body.primary">
            <div class="cell__title">
                {{ cellData.body.title(statute) }}
            </div>
            <div class="cell__usc-label">
                {{ cellData.body.label(statute) }}
            </div>
            <div class="cell__name">
                {{ cellData.body.name(statute) }}
            </div>
        </template>
        <template v-else>
            <a
                v-if="cellData.body.url(statute)"
                :class="cellData.body.type"
                :href="cellData.body.url(statute)"
                target="_blank"
                rel="noopener noreferrer"
            >{{ cellData.body.text(statute) }}</a>
            <span v-else :data-testid="statute.usc + '-none'">None</span>
        </template>
    </td>
</template>
