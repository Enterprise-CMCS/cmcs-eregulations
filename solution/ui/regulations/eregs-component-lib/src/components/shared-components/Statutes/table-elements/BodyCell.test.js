import flushPromises from "flush-promises";
import { render } from "@testing-library/vue";
import { describe, it, expect } from "vitest";

import { ssaSchema } from "../schemas/tableSchemas";
import statutesFixture from "cypress/fixtures/statutes.json";

import BodyCell from "./BodyCell.vue";

describe("Statute Table Body Cell", () => {
    describe("SSA table type", () => {
        ssaSchema.forEach((column, index) => {
            it(`Creates a snapshot of a body cell for column ${index + 1}`, async () => {
                const wrapper = render(BodyCell, {
                    props: {
                        cellData: column,
                        statute: statutesFixture[0],
                    },
                });
                await flushPromises();
                expect(wrapper).toMatchSnapshot();
            });
        });
    });
});
