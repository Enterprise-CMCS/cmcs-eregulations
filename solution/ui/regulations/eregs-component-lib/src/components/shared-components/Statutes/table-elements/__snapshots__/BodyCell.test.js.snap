// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Statute Table Body Cell > SSA table type > Creates a snapshot of a body cell for column 1 1`] = `
{
  "baseElement": <body>
    <div>
      <td
        class="row__cell row__cell--body row__cell--primary"
      >
        <div
          class="cell__title"
        >
           SSA Section 1101 
        </div>
        <div
          class="cell__usc-label"
        >
           42 U.S.C. 1301 
        </div>
        <div
          class="cell__name"
        >
           Definitions 
        </div>
      </td>
    </div>
  </body>,
  "container": <div>
    <td
      class="row__cell row__cell--body row__cell--primary"
    >
      <div
        class="cell__title"
      >
         SSA Section 1101 
      </div>
      <div
        class="cell__usc-label"
      >
         42 U.S.C. 1301 
      </div>
      <div
        class="cell__name"
      >
         Definitions 
      </div>
    </td>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "isUnmounted": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "unmount": [Function],
  "updateProps": [Function],
}
`;

exports[`Statute Table Body Cell > SSA table type > Creates a snapshot of a body cell for column 2 1`] = `
{
  "baseElement": <body>
    <div>
      <td
        class="row__cell row__cell--body row__cell--secondary"
      >
        <a
          class="external"
          href="https://uscode.house.gov/view.xhtml?hl=false&edition=prelim&req=granuleid%3AUSC-prelim-title42-section1301"
          rel="noopener noreferrer"
          target="_blank"
        >
          1301
        </a>
      </td>
    </div>
  </body>,
  "container": <div>
    <td
      class="row__cell row__cell--body row__cell--secondary"
    >
      <a
        class="external"
        href="https://uscode.house.gov/view.xhtml?hl=false&edition=prelim&req=granuleid%3AUSC-prelim-title42-section1301"
        rel="noopener noreferrer"
        target="_blank"
      >
        1301
      </a>
    </td>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "isUnmounted": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "unmount": [Function],
  "updateProps": [Function],
}
`;

exports[`Statute Table Body Cell > SSA table type > Creates a snapshot of a body cell for column 3 1`] = `
{
  "baseElement": <body>
    <div>
      <td
        class="row__cell row__cell--body row__cell--secondary"
      >
        <a
          class="pdf"
          href="https://www.govinfo.gov/content/pkg/COMPS-8763/pdf/COMPS-8763.pdf"
          rel="noopener noreferrer"
          target="_blank"
        >
          Title XI
        </a>
      </td>
    </div>
  </body>,
  "container": <div>
    <td
      class="row__cell row__cell--body row__cell--secondary"
    >
      <a
        class="pdf"
        href="https://www.govinfo.gov/content/pkg/COMPS-8763/pdf/COMPS-8763.pdf"
        rel="noopener noreferrer"
        target="_blank"
      >
        Title XI
      </a>
    </td>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "isUnmounted": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "unmount": [Function],
  "updateProps": [Function],
}
`;

exports[`Statute Table Body Cell > SSA table type > Creates a snapshot of a body cell for column 4 1`] = `
{
  "baseElement": <body>
    <div>
      <td
        class="row__cell row__cell--body row__cell--secondary"
      >
        <a
          class="pdf"
          href="https://www.govinfo.gov/link/uscode/42/1301"
          rel="noopener noreferrer"
          target="_blank"
        >
          1301
        </a>
      </td>
    </div>
  </body>,
  "container": <div>
    <td
      class="row__cell row__cell--body row__cell--secondary"
    >
      <a
        class="pdf"
        href="https://www.govinfo.gov/link/uscode/42/1301"
        rel="noopener noreferrer"
        target="_blank"
      >
        1301
      </a>
    </td>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "isUnmounted": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "unmount": [Function],
  "updateProps": [Function],
}
`;

exports[`Statute Table Body Cell > SSA table type > Creates a snapshot of a body cell for column 5 1`] = `
{
  "baseElement": <body>
    <div>
      <td
        class="row__cell row__cell--body row__cell--secondary"
      >
        <a
          class="external"
          href="https://www.ssa.gov/OP_Home/ssact/title11/1101.htm"
          rel="noopener noreferrer"
          target="_blank"
        >
          1101
        </a>
      </td>
    </div>
  </body>,
  "container": <div>
    <td
      class="row__cell row__cell--body row__cell--secondary"
    >
      <a
        class="external"
        href="https://www.ssa.gov/OP_Home/ssact/title11/1101.htm"
        rel="noopener noreferrer"
        target="_blank"
      >
        1101
      </a>
    </td>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "isUnmounted": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "unmount": [Function],
  "updateProps": [Function],
}
`;

exports[`Statute Table Body Cell > SSA table type > displays 'None' in Statute Compilation BodyCell when source_url is null 1`] = `
{
  "baseElement": <body>
    <div>
      <td
        class="row__cell row__cell--body row__cell--secondary"
      >
        <span
          data-testid="1302-none"
        >
          None
        </span>
      </td>
    </div>
  </body>,
  "container": <div>
    <td
      class="row__cell row__cell--body row__cell--secondary"
    >
      <span
        data-testid="1302-none"
      >
        None
      </span>
    </td>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "isUnmounted": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "unmount": [Function],
  "updateProps": [Function],
}
`;
